src/parser/GenericParser.ts

```diff
@@ -114,17 +94,32 @@ export class GenericParser {
 
             // Await completion of file metric calculations:
             const promisesResults = await Promise.all(fileMetricPromises);
-            for (const [filepath, fileMetricResults] of promisesResults) {
-                fileMetrics.set(filepath, fileMetricResults);
-            }
 
-            for (let i = 0; i < unsupportedFiles.length; i++) {
-                unsupportedFiles[i] = formatPrintPath(unsupportedFiles[i], this.config);
+            for (const [sourceFile, fileMetricResults] of promisesResults) {
+                const printPath = formatPrintPath(sourceFile.filePath, this.config);
+                if (!isErrorFile(sourceFile)) {
+                    fileMetrics.set(printPath, fileMetricResults);
+
+                    // Inform about errors that occurred while calculating (some) metrics on the syntax tree
+                    for (const metricError of fileMetricResults.metricErrors.values()) {
+                        console.error(
+                            "Error while calculating the metric " +
+                                metricError.metricName +
+                                " on the file " +
+                                sourceFile.filePath,
+                        );
+                        console.error(metricError.error);
+                    }
+                    if (!isParsedFile(sourceFile)) {
+                        unsupportedFiles.push(formatPrintPath(sourceFile.filePath, this.config));
```

Comment:
don't calculate print path again
```suggestion
                        unsupportedFiles.push(printPath);
```